pkg/controller/handlers/workflowstep/loop.go

```diff
@@ -146,66 +137,80 @@ func elementPrompt(element, prompt string) string {
 	`, element, prompt)
 }
 
-func defineDataStep(rootStep *v1.WorkflowStep) *v1.WorkflowStep {
+func defineDataStep(rootStep *v1.WorkflowStep, fileName string) *v1.WorkflowStep {
 	return NewStep(rootStep.Namespace, rootStep.Spec.WorkflowExecutionName, rootStep.Spec.AfterWorkflowStepName, rootStep.Spec.WorkflowGeneration, types.Step{
-		ID:   rootStep.Spec.Step.ID + "-loopdata",
-		Step: dataPrompt(rootStep.Spec.Step.Step),
+		ID:   rootStep.Spec.Step.ID + "{loopdata}",
```

Comment:
This is a fine hack, though I worry what trouble this might produce.

Reply:
So for this I am following the older pattern of how we did things for while loops and if statements. For "substeps" generated by the top-level if- and while-steps, there would be additional information surrounded by braces. There is code in the UI that expects this as well. So I'm just trying to follow that convention here.